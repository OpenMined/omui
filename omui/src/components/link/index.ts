export { Link } from './link';
export { default as LinkStyle } from './link.style';
