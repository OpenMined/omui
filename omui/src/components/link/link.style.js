export default {
  baseStyle: {
    transition: `all 0.15s ease-out`,
    cursor: 'pointer',
    textDecoration: 'none',
    outline: 'none',
    color: 'blue.500',
    _hover: {
      color: 'blue.700'
    },
    _disabled: {
      opacity: 0.4,
      cursor: 'not-allowed',
      textDecoration: 'none'
    }
  }
};
