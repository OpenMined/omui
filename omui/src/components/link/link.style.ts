// import { ComponentTheme } from '@chakra-ui/theme-tools';

// TODO: remove the 'any' type
const Link: any = {
  baseStyle: {
    transition: `all 0.15s ease-out`,
    cursor: 'pointer',
    textDecoration: 'none',
    outline: 'none',
    color: 'inherit',
    _hover: {
      color: 'blue.700'
    },
    _disabled: {
      opacity: 0.4,
      cursor: 'not-allowed',
      textDecoration: 'none'
    }
  }
};

export default Link;
