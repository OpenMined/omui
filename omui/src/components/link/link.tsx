import React from 'react';
import { Link as ChakraLink, LinkProps } from '@chakra-ui/core';
import { ExternalLinkIcon } from '@chakra-ui/icons';

export const Link = ({ isExternal, children, ...props }: LinkProps) => {
  if (
    props.href &&
    (props.href.includes('http://') || props.href.includes('https://'))
  ) {
    isExternal = true;
  }

  return (
    <ChakraLink {...props} isExternal={isExternal}>
      {children}
      {isExternal && <ExternalLinkIcon ml={1} mt="-4px" />}
    </ChakraLink>
  );
};
