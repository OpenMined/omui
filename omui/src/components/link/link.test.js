import React from 'react';
import renderer from 'react-test-renderer';
import { Link } from './link';

it('renders correctly with external ref', () => {
  const tree = renderer
    .create(<Link href="https://openmined.org">OpenMined</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with internal ref', () => {
  const tree = renderer.create(<Link href="#cards">Cards</Link>).toJSON();
  expect(tree).toMatchSnapshot();
});
