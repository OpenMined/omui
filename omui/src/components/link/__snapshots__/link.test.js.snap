// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with external ref 1`] = `
<a
  className="chakra-link css-0"
  href="https://openmined.org"
  rel="noopener noreferrer"
  target="_blank"
>
  OpenMined
  <svg
    className="chakra-icon css-1hxuad4"
    focusable={false}
    viewBox="0 0 24 24"
  >
    <g
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeWidth="2"
    >
      <path
        d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
      />
      <path
        d="M15 3h6v6"
      />
      <path
        d="M10 14L21 3"
      />
    </g>
  </svg>
</a>
`;

exports[`renders correctly with internal ref 1`] = `
<a
  className="chakra-link css-0"
  href="#cards"
>
  Cards
</a>
`;
